### PR TITLE
refactor to use again the flipletLogin passport

### PR DESCRIPTION
**Pull request by squallstar**
Fixes an issue where the login component thinks users are already logged in when they're not.

The issue was recently discovered in the **NWP Admin Web App**.

---

ref https://github.com/Fliplet/fliplet-studio/issues/2847

### CODE REVIEW

File: js/build.js [hunk 1]
```diff
@@ -372,22 +372,32 @@ $('[data-login-id]').each(function() {
...
-          id: session.user.id,
+          id: loggedUser.id || session.user.id,
           region: session.auth_token.substr(0, 2),
-          userRoleId: session.user.userRoleId,
+          userRoleId: loggedUser.userRoleId || session.user.userRoleId,
           authToken: session.auth_token,
-          email: session.user.email,
-          legacy: session.legacy
+          email: loggedUser.email || session.user.email,
+          legacy: loggedUser.legacy || session.legacy
```
Comment (tonytlwu) [inline js/build.js]: @squallstar Can you verify if the legacy property is non-empty when logging in with an account with a legacy v1 user? If it's empty then Fliplet Viewer will crash when opening v1 apps.

Comment (squallstar) [reply js/build.js]: @tonytlwu will do, is this for FV 2.0? If so, my guess is I need to rename `loggedUser.legacy` to `legacyId` and make it a boolean

Comment (tonytlwu) [reply js/build.js]: @squallstar `legacy` is expected to be an object

https://github.com/Fliplet/fliplet-ios-version-2/blob/f3b74606b74672206a91e19dadec8c79685b8b74/Fliplet/Classes/FLTAuthenticationManager.m#L50-L54

Comment (tonytlwu) [reply js/build.js]: It might require a server side change because that could be why I was using `session.user`. It's all coming back to me now 🥇

Comment (squallstar) [reply js/build.js]: ahhh thanks for the heads up, @tonytlwu yeah I think it will need a server-side change. I'll look into this.

**Pull request merged**